### ISSUE TITLE
[import/named] will report error upon "export * from"

### ISSUE THREAD

**Issue by xsjcTony** [import-js/eslint-plugin-import#2537]
First of all, I've tried search through issues and google, but cannot find anything helpful, if I miss something, sorry for that.

---

If I import something from a module with `export * from anotherModule`, the `import/named` rule will throw an error.

Example:
```typescript
// a.ts
export const foo = 'bar'
```
```typescript
// b.ts
export * from './a'
```
```typescript
// c.ts
import { foo } from './b' // import/named: foo not found in './b.ts'
```

I'm not sure if this is a false positive or intended, is there any workaround?

---
In `Vue3`, this happens a lot, for example:
```vue
<script setup lang="ts">
import { ref } from 'vue' // import/named: ref not found in 'vue'
</script>
```
This happens because in `vue.d.ts`, `ref` is exported by `export * from "@vue/runtime-dom"`

---

Version: (In a word, all of them are the latest)
| Package | Version |
| ----------- | ----------- |
| eslint | `8.22.0` |
| typescript | `4.8.2`  |
| eslint-plugin-import | `2.26.0` |
| eslint-import-resolver-node | `0.3.6` |
|eslint-import-resolver-typescript|`3.5.0`|

Node.JS: `16.16.0`

**Comment by ljharb** [comment]
What version of the plugin are you using?

**Comment by xsjcTony** [comment]
> What version of the plugin are you using?

Sorry I forgot to mention it. I've updated the issue

**Comment by ljharb** [comment]
This seems like it might be due to the Vue eslint parser I assume you're using - we already have passing tests for this exact scenario: https://github.com/import-js/eslint-plugin-import/blob/04e114b686bd5acab887ebacaa3b178ea264c96a/tests/src/rules/named.js#L369-L382

**Comment by xsjcTony** [comment]
Not sure about that. Since the same thing happens for one of my `React` app which uses `@typescript-eslint/parser` as parser.

https://github.com/xsjcTony/wordle
(Sorry for the inconvenience, as I have no idea how to apply ESLint on platforms like StackBlitz)
You can clone this repo and import something from `/src/test/utils` which is in `@testing-library/react` such as `screen`.
`import { screen } from '@/test/utils'` and the error should be raised.
And please don't forget to turn on the `import/named` rule here:
https://github.com/xsjcTony/wordle/blob/4524cd8022583674871ef62d561f1be9cb6a11f1/.eslintrc.cjs#L56

**Comment by ljharb** [comment]
Could you make a branch that already has the needed changes, so i can clone that directly?

**Comment by xsjcTony** [comment]
I'm sorry that I cannot reproduce it now. I'm just so confused. :-<
I was pretty sure the error was raised otherwise I wouldn't turn the rule off.

I guess I'll just turn the rule off for now in my `Vue` app, and if I meet the problem again with a normal parser I'll immediately provide a reproducible repo.

**Comment by ljharb** [comment]
Thanks, appreciate it. A repro with the vue parser might still help, fwiw. I'll close for now.

**Comment by xsjcTony** [comment]
https://github.com/xsjcTony/aelita.me

@ljharb Hi, Here's a repro with both `@typescript-eslint/parser` and `vue parser`

###  `typescript-eslint/parser`:
Clone the repo and run `pnpm install` or npm, yarn whatever, the error should be raised straight away in IDE, and in `pnpm run dev` as well
![Snipaste_2022-08-30_05-16-57](https://user-images.githubusercontent.com/45784210/187280668-14aaff48-d8be-4347-8f39-4b7c65f3b1e4.png)
![Snipaste_2022-08-30_05-17-10](https://user-images.githubusercontent.com/45784210/187280718-74480374-5b3e-4c20-9f33-fb797c43e220.png)

### `vue parser`:
Delete the `.eslintrc.cjs` file and rename `.eslintrcvue.cjs` to `.eslintrc.cjs`.
Now both the previous one and the new one in `/src/components/HelloWorld.vue` should raise error.
![Snipaste_2022-08-30_05-19-36](https://user-images.githubusercontent.com/45784210/187281078-2a6c87e1-4bef-493d-aa9f-1639ed003994.png)

**Comment by ljharb** [comment]
The error is not raised with `npm run dev`, or with `npx eslint .` or `npx eslint src/main.ts` or `npx eslint --config=.eslintrcvue.cjs src/main.ts`.

**Comment by xsjcTony** [comment]
No error for both parsers? emm... I'm way so confused.
I guess it's either something wrong with my environment or just something weird that happened.
If it's not reproducible then no way we can solve it.

Anyway thanks a lot for taking care of my issue. Really appreciated.

**Comment by ljharb** [comment]
To be clear, i'm using `npm install` - pnpm and yarn's PnP often break things, so if you can repro it with those but not with npm, then i'd file an issue there.

**Comment by xsjcTony** [comment]
You are so smart, sir.
There is no problem with `npm`.
From my experience of my last react app which was mentioned above, I'm guessing the problem may just go away several days later with `pnpm` for whatever reasons.
I have to switch to `npm` now, but it's soooo slow compare to `pnpm`

Thanks a lot again!🤝

**Comment by ljharb** [comment]
That can be frustrating, but nothing's slower than incorrect ¯\\\_(ツ)_/¯ glad npm is working for you
